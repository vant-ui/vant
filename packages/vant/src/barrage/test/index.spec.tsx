import { ref } from 'vue';
import { Barrage, BarrageInstance } from '..';
import { mount, trigger } from '../../../test';
import { flushPromises } from '@vue/test-utils';

test('should auto play when only list props', async () => {
  const wrapper = mount({
    render() {
      return (
        <Barrage
          barrageList={[
            '轻量',
            '可定制的',
            '移动端',
            'Vue',
            '组件库',
            'VantUI',
            '666',
          ]}
        >
          <div class="video" style="width: 100%; height: 150px"></div>
        </Barrage>
      );
    },
  });

  await flushPromises();
  expect(wrapper.html()).toMatchSnapshot();
  expect(wrapper.findAll('.van-barrage__item')).toHaveLength(7);
});

test('should not auto play use play function when use play function', async () => {
  const barrage = ref<BarrageInstance>();
  const wrapper = mount({
    render() {
      return (
        <Barrage
          ref={barrage}
          autoPlay={false}
          barrageList={[
            '轻量',
            '可定制的',
            '移动端',
            'Vue',
            '组件库',
            'VantUI',
            '666',
          ]}
        >
          <div class="video" style="width: 100%; height: 150px"></div>
        </Barrage>
      );
    },
  });

  await flushPromises();
  expect(wrapper.html()).toMatchSnapshot();
  expect(wrapper.findAll('.van-barrage__item')).toHaveLength(7);
  expect(
    (wrapper.find('.van-barrage__item') as HTMLSpanElement).style
      .animationPlayState
  ).toBe('paused');

  barrage.value?.play();

  expect(
    (wrapper.find('.van-barrage__item') as HTMLSpanElement).style
      .animationPlayState
  ).toBe('running');

  barrage.value?.pause();

  expect(
    (wrapper.find('.van-barrage__item') as HTMLSpanElement).style
      .animationPlayState
  ).toBe('paused');

  barrage.value?.play();
  barrage.value?.add('Barrage');
  expect(wrapper.findAll('.van-barrage__item')).toHaveLength(8);

  trigger(
    wrapper.find('.van-barrage__item') as HTMLSpanElement,
    'animationend'
  );
  expect(wrapper.findAll('.van-barrage__item')).toHaveLength(7);
});
