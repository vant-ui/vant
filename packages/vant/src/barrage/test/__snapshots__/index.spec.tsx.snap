// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should auto play when only list props 1`] = `
<div class="van-barrage">
  <div class="video"
       style="width: 100%; height: 150px;"
  >
  </div>
</div>
`;

exports[`should not auto play use play function when use play function 1`] = `
<div class="van-barrage">
  <div class="video"
       style="width: 100%; height: 150px;"
  >
  </div>
</div>
`;
